src/systems/combat.rs

```diff
@@ -0,0 +1,202 @@
+use amethyst::core::Time;
+use amethyst::shrev::{EventChannel, ReaderId};
+use amethyst::{core::Transform, ecs::*, Error};
+use amethyst_test::AmethystApplication;
+
+use crate::components::combat;
+use crate::components::combat::{Cooldown, Damage, Speed};
+use crate::components::creatures::{CarnivoreTag, HerbivoreTag, PlantTag};
+use crate::components::health::Health;
+use crate::systems::collision::CollisionEvent;
+use std::time::Duration;
+
+pub struct CooldownSystem;
+
+impl<'s> System<'s> for CooldownSystem {
+    type SystemData = (WriteStorage<'s, Cooldown>, Entities<'s>, Read<'s, Time>);
+
+    fn run(&mut self, (mut cooldowns, entities, time): Self::SystemData) {
+        let mut to_remove = Vec::new();
+
+        for (mut cooldown, entity) in (&mut cooldowns, &*entities).join() {
+            match cooldown.time_left.checked_sub(time.delta_time()) {
```

Comment:
I would only change `delta_time` to `fixed_time`, delta_time will make simulations depend on the frame-rate stability, making the simulation non-deterministic, which makes debugging harder and we lose a possible feature.

```suggestion
            match cooldown.time_left.checked_sub(time.fixed_time()) {
```

Reply:
Relevant docs: https://amethyst.rs/doc/latest/doc/amethyst_core/timing/struct.time#method.fixed_time

Reply:
@khskarl I think `fixed_time` is not affected by `time_scale`. I think it would also be problematic because `delta_time` would reflect the time between two computations, while `fixed_time` is constant, with an unlimited frame rate the results would change a lot.

I'll merge it for now and maybe we can revisit this topic when we decide to have multiple systems for different domains (e.g. one system group for simulations and one for effects)